.github/workflows/publish.yml

```
@@ -0,0 +1,49 @@
+name: Publish to PyPI
+
+on:
+  push:
+    branches:
+      - main
+    tags:
+      - "*"
+
+jobs:
+  build:
+    name: Build package
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v4
+
+    - name: Set up Python 3.12
+      uses: actions/setup-python@v5
+      with:
+        python-version: 3.12
+
+    - name: Install dependencies
+      run: |
+        if [ -f requirements.txt ]; then pip install -r requirements.txt; fi
+
+    - name: Build package
+      run: python3 -m build
+
+    - name: Upload build
+      uses: actions/upload-artifact@v4
+      with:
+        path: dist/*
```

May as well throw here if the files are not found:
```suggestion
        path: dist/*
        if-no-files-found: error
```

https://github.com/actions/upload-artifact?tab=readme-ov-file#customization-if-no-files-are-found